src/instance_check.jl

```diff
@@ -4,29 +4,31 @@
 const HAS_OPENING_TIME_WINDOWS = 1
 const HAS_MULTIPLE_OPENING_TIME_WINDOWS = 2
 const HAS_FLEXIBLE_OPENING_TIME_WINDOWS = 3
+const HAS_A_RECHARGING_STATION = 4
 
 # Product based features #
-const HAS_PRODUCT_CONFLICT_CLASSES = 4
-const HAS_PRODUCT_PROHIBITED_PREDECESSOR_CLASSES = 5
-const HAS_PRODUCT_PICKUPONLY_SHARING_CLASSES = 6
-const HAS_PRODUCT_DELIVERYONLY_SHARING_CLASSES = 7
-const HAS_PRODUCT_SHIPMENT_SHARING_CLASSES = 8
+const HAS_PRODUCT_COMPATIBILITY_CONFLICT_ISSUES = 10
+const HAS_PRODUCT_PROHIBITED_PREDECESSOR_ISSUES = 11
+const HAS_PRODUCT_LIMITED_AVAILABILITIES_AT_PICKUP_POINTS = 12
+const HAS_PRODUCT_LIMITED_CAPACITIES_AT_DELIVERY_POINTS = 13
 
 # Request based features
 const HAS_SHIPMENT_REQUESTS = 109
 const HAS_PICKUPONLY_REQUESTS = 110
 const HAS_DELIVERYONLY_REQUESTS = 111
-const HAS_ALTERNATIVE_PICKUP_LOCATIONS = 112 #
-const HAS_ALTERNATIVE_DELIVERY_LOCATIONS = 113 #
-const HAS_MAX_DURATION = 114
-const HAS_DURATION_UNIT_COSTS = 115 #
-const HAS_PICKUP_TIME_WINDOWS = 116
-const HAS_DELIVERY_TIME_WINDOWS = 117
-const HAS_MULTIPLE_PICKUP_TIME_WINDOWS = 118
-const HAS_MULTIPLE_DELIVERY_TIME_WINDOWS = 119
-const HAS_FLEXIBLE_PICKUP_TIME_WINDOWS = 120 #
-const HAS_FLEXIBLE_DELIVERY_TIME_WINDOWS = 121 #
+const HAS_SHARED_PRODUCTS_DELIVERIES_BEING_SATISFIED_FROM_PICKUPS = 112
+const HAS_ALTERNATIVE_PICKUP_LOCATIONS_FOR_REQUESTS = 112 #
```

Comment:
Are there (or will there be) pick_locations/delivery_locations for something other than requests? My question also applies to most of the following renamings. 

For those that do not have ambiguous meanings, I prefer not to use the longer version of names since it will require us to split several lines in the code to 2 lines when respecting the 80 char line limit, and this will reduce readability. 

For those that do have ambiguous meaning such as MAX_DURATION, adding REQUEST is good to not require the developer to refer to the comments in the code or the documentation.

Reply:
All your suggestions suits me well